include/vcpkg/base/expected.h

```diff
@@ -14,6 +14,26 @@
 
 namespace vcpkg
 {
+    struct Unit
+    {
+        // A meaningless type intended to be used with Expected when there is no meaningful value.
+    };
```

Comment:
Why no `Expected<void>`?

Reply:
We decided to not do the "duplicate all the code in expected" solution in https://github.com/microsoft/vcpkg-tool/pull/564